tests/functional/src/main/java/org/mule/functional/junit4/ExtensionsTestInfrastructureDiscoverer.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.junit4;
+
+import static org.mule.runtime.core.config.MuleManifest.getProductVersion;
+import org.mule.runtime.core.DefaultMuleContext;
+import org.mule.runtime.core.api.lifecycle.InitialisationException;
+import org.mule.runtime.core.api.registry.MuleRegistry;
+import org.mule.runtime.core.api.registry.ServiceRegistry;
+import org.mule.runtime.core.config.MuleManifest;
+import org.mule.runtime.core.registry.DefaultRegistryBroker;
+import org.mule.runtime.core.registry.MuleRegistryHelper;
+import org.mule.runtime.core.registry.SpiServiceRegistry;
+import org.mule.runtime.core.util.ArrayUtils;
+import org.mule.runtime.core.util.collection.ImmutableListCollector;
+import org.mule.runtime.extension.api.ExtensionManager;
+import org.mule.runtime.extension.api.introspection.ExtensionFactory;
+import org.mule.runtime.extension.api.introspection.declaration.DescribingContext;
+import org.mule.runtime.extension.api.introspection.declaration.spi.Describer;
+import org.mule.runtime.extension.api.resources.GeneratedResource;
+import org.mule.runtime.extension.api.resources.ResourcesGenerator;
+import org.mule.runtime.extension.api.resources.spi.GeneratedResourceFactory;
+import org.mule.runtime.module.extension.internal.DefaultDescribingContext;
+import org.mule.runtime.module.extension.internal.introspection.DefaultExtensionFactory;
+import org.mule.runtime.module.extension.internal.introspection.describer.AnnotationsBasedDescriber;
+import org.mule.runtime.module.extension.internal.introspection.version.StaticVersionResolver;
+import org.mule.runtime.module.extension.internal.manager.DefaultExtensionManager;
+import org.mule.runtime.module.extension.internal.manager.ExtensionManagerAdapter;
+import org.mule.runtime.module.extension.internal.resources.AbstractResourcesGenerator;
+
+import com.google.common.collect.ImmutableList;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.jar.Manifest;
+
+import org.apache.commons.io.FileUtils;
+
+/**
+ * It creates a {@link ExtensionManager} and automatically registers extensions.
```

Comment:
The javadoc does not seem to be correct. There is a constructor that receives the extensionManager. It would be more testable if both constructors receive it as a parameter.